cmd/clusterctl/client/upgrade.go

```diff
@@ -99,6 +99,14 @@ func (c *clusterctlClient) ApplyUpgrade(options ApplyUpgradeOptions) error {
 		return err
 	}
 
+	// Ensures the latest version of cert-manager.
+	// NOTE: it is safe to upgrade to latest version of cert-manager given that it provides
+	// conversion web-hooks around Issuer/Certificate kinds, so installing an older versions of providers
+	// should continue to work with the latest cert-manager.
+	if err := clusterClient.CertManager().EnsureLatestVersion(); err != nil {
```

Comment:
We should do this after `parseUpgradeItem`.

As I was testing this out, I had done `clusterctl upgrade apply` (that is, without any `management-group` or `contract` flags) and it started to upgrade cert-manager and after it did, I got the following error.
```
Error: invalid provider name "". Provider name should be in the form namespace/provider[:version]
```

It would be nice to have the flag validation done before any upgrading actually takes place. Else the operator may be confused that the cluster is in a "half-upgraded" state.

Reply:
makes sense